onnx/common/ir.h

```diff
@@ -671,6 +671,23 @@ struct Node : public Attributes<Node> {
     inputs_.clear();
   }
 
+  // Check whether this node is before node n in the graph.
+  bool isBefore(Node* n) {
+    if (n == nullptr || this == n) {
+      // Bail out early.
+      return false;
+    }
+    ONNX_ASSERT(n->inGraphList());
+    Node* p = next();
+    while (p != nullptr) {
```

Comment:
you're going to get featured in https://accidentallyquadratic.tumblr.com/

but it's probably fine in this case